js/signup.js

```diff
@@ -0,0 +1,195 @@
+const querySelector = (selector) => document.querySelector(selector);
+
+const emailInput = querySelector("#email-input");
+const passwordInput = querySelector("#password-input");
+const emailErrorMessage = querySelector("#email-error-message");
+const passwordErrorMessage = querySelector("#password-error-message");
+const passwordConfirmInput = querySelector("#password-confirm-input");
+const passwordConfirmErrorMessage = querySelector("#password-confirm-error-message");
+const loginForm = querySelector("#login-form");
+
+const eyeOffButton = querySelector("#eye-off");
+const eyeOnButton = querySelector("#eye-on");
+const eyeOffConfirmButton = querySelector("#eye-off-confirm");
+const eyeOnConfirmButton = querySelector("#eye-on-confirm");
+
+const EMAIL_REGEX = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$/;
+const PASSWORD_REGEX = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
+
+// 이메일 입력
+emailInput.addEventListener("focusout", validateEmailInput);
+
+function isValidateEmail(email) {
+  return EMAIL_REGEX.test(email);
+}
+
+function validateEmailInput() {
+  const email = emailInput.value.trim();
+  const testEmail = "test@codeit.com";
+
+  if (email === "") {
+    showEmailErrorMessage("이메일을 입력해 주세요.");
+    return;
+  }
+
+  if (!isValidateEmail(email)) {
+    showEmailErrorMessage("올바른 이메일 주소가 아닙니다.");
+    return;
+  }
+
+  if (email === testEmail) {
+    showEmailErrorMessage("이미 사용 중인 이메일입니다.");
+    return;
+  }
+
+  hideEmailErrorMessage();
+}
+
+function showEmailErrorMessage(message) {
+  emailErrorMessage.textContent = message;
+  emailErrorMessage.classList.add("show");
+  emailInput.classList.add("sign-input-error");
+  emailInput.classList.add("error-focus");
+}
+
+function hideEmailErrorMessage() {
+  emailErrorMessage.textContent = "";
+  emailErrorMessage.classList.remove("show");
+  emailInput.classList.remove("sign-input-error");
+  emailInput.classList.remove("error-focus");
+}
+
+// 비밀번호 입력
+passwordInput.addEventListener("focusout", validatePasswordInput);
+
+function isValidatePassword(password) {
```

Comment:
`validate` -> `valid` 요렇게 되면 좋을 것 같아요!
```suggestion
function isValidPassword(password) {
```